multi-node/aws/pkg/config/config.go

```diff
@@ -227,6 +228,25 @@ func (c Cluster) stackConfig(opts StackTemplateOptions, compressUserData bool) (
 
 	stackConfig.Config.TLSConfig = compactAssets
 
+	controllerIPAddr := net.ParseIP(stackConfig.ControllerIP)
+	if controllerIPAddr == nil {
+		return nil, fmt.Errorf("invalid controllerIP: %s", stackConfig.ControllerIP)
+	}
+	controllerSubnetFound := false
+	lastSubnetIndex := len(stackConfig.Subnets) - 1
+	for i, subnet := range stackConfig.Subnets {
+		_, instanceCIDR, err := net.ParseCIDR(subnet.InstanceCIDR)
+		if err != nil {
+			return nil, fmt.Errorf("invalid instanceCIDR: %v", err)
+		}
+		if instanceCIDR.Contains(controllerIPAddr) {
```

Comment:
I see that my prior comment about "still checking controller is in first subnet" has already been fixed! Sorry, I blame github for being weird ;)